src/Common/src/TypeSystem/IL/Stubs/ILEmitter.cs

```diff
@@ -201,6 +204,189 @@ public void EmitSwitch(ILCodeLabel[] labels)
             }
         }
 
+        public void EmitLdInd(TypeDesc type)
+        {
+            switch (type.Category)
+            {
+                case TypeFlags.Byte:
+                case TypeFlags.SByte:
+                case TypeFlags.Boolean:
+                    Emit(ILOpcode.ldind_i1);
+                    break;
+                case TypeFlags.Char:
+                case TypeFlags.UInt16:
+                case TypeFlags.Int16:
+                    Emit(ILOpcode.ldind_i2);
+                    break;
+                case TypeFlags.UInt32:
+                case TypeFlags.Int32:
+                    Emit(ILOpcode.ldind_i4);
+                    break;
+                case TypeFlags.UInt64:
+                case TypeFlags.Int64:
+                    Emit(ILOpcode.ldind_i8);
+                    break;
+                case TypeFlags.Single:
+                    Emit(ILOpcode.ldind_r4);
+                    break;
+                case TypeFlags.Double:
+                    Emit(ILOpcode.ldind_r8);
+                    break;
+                case TypeFlags.IntPtr:
+                case TypeFlags.UIntPtr:
+                case TypeFlags.Pointer:
+                case TypeFlags.FunctionPointer:
+                    Emit(ILOpcode.ldind_i);
+                    break;
+                case TypeFlags.Array:
+                case TypeFlags.SzArray:
+                case TypeFlags.Class:
+                case TypeFlags.Interface:
+                    Emit(ILOpcode.ldind_ref);
+                    break;
+                default:
+                    Emit(ILOpcode.ldobj, _emitter.NewToken(type));
```

Comment:
Could you add an assert that the flags are not one of the never valid values? `Void`, `ByRef`, `GenericParameter`, `SignatureTypeVariable`, `SignatureMethodVariable` should be the full list. It's probably better to catch it here than to troubleshoot RyuJIT asserts later.